mmeval/metrics/coco_panoptic.py

```diff
@@ -0,0 +1,699 @@
+        classwise (bool): Whether to evaluate the metric class-wise.
+            Defaults to False.
+        outfile_prefix (str, optional): The prefix of json files. It includes
+            the file path and the prefix of filename, e.g., "a/b/prefix".
+            If not specified, a temp file will be created.
+            It should be specified when format_only is True. Defaults to None.
+        nproc (int): Number of processes for panoptic quality computing.
+            Defaults to 4. When ``nproc`` exceeds the number of cpu cores,
+            the number of cpu cores is used.
+        backend_args (dict, optional): Arguments to instantiate the
+            preifx of uri corresponding backend. Defaults to None.
+        categories (list, optional): The list of catetories of panoramic segmentation.
+    """
+    def __init__(self,
+                 ann_file: Optional[str] = None,
+                 seg_prefix: Optional[str] = None,
+                 classwise: bool = False,
+                 nproc: int = 4,
+                 outfile_prefix: Optional[str] = None,
+                 backend_args: Optional[dict] = None,
+                 categories: list = None,
+                 **kwargs) -> None:
+        if panopticapi is None:
+            raise RuntimeError(
+                'panopticapi is not installed, please install it by: '
+                'pip install git+https://github.com/cocodataset/'
+                'panopticapi.git.')
+        super().__init__(**kwargs)
+        self.classwise = classwise
+        self.tmp_dir = None
+        # outfile_prefix should be a prefix of a path which points to a shared
+        # storage when train or test with multi nodes.
+        self.outfile_prefix = outfile_prefix
+        if outfile_prefix is None:
+            self.tmp_dir = tempfile.TemporaryDirectory()
+            self.outfile_prefix = self.tmp_dir.name 
+        
+        self.nproc = min(nproc, multiprocessing.cpu_count())
+        self.categories = categories  
+
+        if ann_file is not None:
+            with get_local_path(
+                    filepath=ann_file,
+                    backend_args=backend_args) as local_path:
+                self._coco_api = COCOPanoptic(local_path)
+            self.categories = self._coco_api.cats
+        else:
+            self._coco_api = None
+        
+        if self.categories is None:
+            # use defalut coco categories
+            self.categories = []
+            idx = 1
+            for name in CLASSES:
+                if name in STUFF_CLASSES:
+                    isthing = 0
+                else:
+                    isthing = 1
+                while(idx in MISSING_ID): idx += 1
+                cat_item = {
+                    "supercategory": None,
+                    "isthing": isthing,
+                    "id": idx,
+                    "name" : name
+                }
+                idx += 1
+                self.categories.append(cat_item)
```

Comment:
Why do we create `self.categories` manually instead of accessing through `self._coco_api`~